File: aoc22/java/src/test/java/tech/dobler/aoc22/Day03Test.java
package tech.dobler.aoc22;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.stream.Stream;

import static org.assertj.core.api.Assertions.assertThat;

class Day03Test {

    final String testInput = """
            vJrwpWtwJgWrhcsFMMfFFhFp
            jqHRNqRjqzjGDLGLrsFMfFZSrLrFZsSL
            PmmdzqPrVvPwwTWBwg
            wMqvLMZHhHMvwLHjbvcjnnSBnvTQFn
            ttgJtRGJQctTZtZT
            CrZsJsPPZsGzwwsLwLmpwMDw""";
    private Day03 day;

    @BeforeEach
    void setUp() {
        day = new Day03();
    }

    private static Stream<String> readInput()
    {
        return Util.readLines(3);
    }

    @Test
    void part1() {
        final var result = day.part1(Util.splitByNewLine(testInput));

        assertThat(result).isEqualTo(157);

        final var contents = readInput();
        assertThat(day.part1(contents)).isEqualTo(7_763);
    }

    @Test
    void part2() {
        final var result = day.part2(testInput);

        assertThat(result).isEqualTo(70);

        final var contents = Util.readInput(3);
        assertThat(day.part2(contents)).isEqualTo(2_569);
    }
}

File: aoc21/Kotlin/src/test/kotlin/day03/Day03Test.kt
package day03

import org.junit.jupiter.api.Test
import readInput
import kotlin.test.assertEquals

class Day03Test {

    private val path = "03_test"

    @Test
    fun `parse input`() {
        val input = readInput(path, ::mapInput);
        val expected: Input = listOf(
            listOf(0, 0, 1, 0, 0),
            listOf(1, 1, 1, 1, 0),
            listOf(1, 0, 1, 1, 0),
            listOf(1, 0, 1, 1, 1),
            listOf(1, 0, 1, 0, 1),
            listOf(0, 1, 1, 1, 1),
            listOf(0, 0, 1, 1, 1),
            listOf(1, 1, 1, 0, 0),
            listOf(1, 0, 0, 0, 0),
            listOf(1, 1, 0, 0, 1),
            listOf(0, 0, 0, 1, 0),
            listOf(0, 1, 0, 1, 0)
        )
        assertEquals(expected, input)
    }

    @Test
    fun `part 1 example`()
    {
        val input = readInput(path, ::mapInput);
        assertEquals(198, part1(input))
    }

    @Test
    fun `part 2 example`()
    {
        val input = readInput(path, ::mapInput);
        assertEquals(230, part2(input))
    }
}